redbot/cogs/admin/admin.py

```diff
@@ -421,10 +432,20 @@ def pass_user_hierarchy_check(ctx: commands.Context, role: discord.Role) -> bool
 
         NOTE: The role is case sensitive!
         """
+        if not self.pass_user_hierarchy_check(ctx, role):
+            await ctx.send(
+                _(
+                    "I cannot let you remove {role.name} from being a selfrole because that role is higher than or equal to your highest role in the Discord hierarchy."
+                ).format(role=role)
+            )
+            return
         async with self.conf.guild(ctx.guild).selfroles() as curr_selfroles:
-            curr_selfroles.remove(role.id)
+            if role.id in curr_selfroles:
+                curr_selfroles.remove(role.id)
```

Comment:
This is redundant because we never reach the code due to the SelfRole converter. The old method should have worked fine here.